### PR TITLE
[SchemaRegistry] handle unknown content types

### PR DESCRIPTION
**Pull request by swathipil**
As per discussion with architects, return the content type string in SchemaProperties.format if unknown. Make sure the behavior is well-documented; added a warning to the get_schema method for this behavior.

TODO:
- [x] add a test with MockResponse for an unknown content type.

### CODE REVIEW

**Review by l0lawrence — APPROVED** [review]
lgtm

File: sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md [hunk 1]
```diff
@@ -6,15 +6,16 @@ This version and all future versions will require Python 3.8+. Python 3.7 is no
 
 ### Features Added
 
-- `V2022_10` has been added to `ApiVersion` and set as the default API version.
-  - `Json` and `Custom` have been added to supported formats in `SchemaFormat`.
 - Sync and async `JsonSchemaEncoder` have been added under `azure.schemaregistry.encoder.jsonencoder`.
 - `InvalidContentError` have been added under `azure.schemaregistry.encoder.jsonencoder` for use with the `JsonSchemaEncoder`.
 - `MessageContent`, `OutboundMessageContent`,`InboundMessageContent`, and `SchemaContentValidate` have been added under `azure.schemaregistry` as protocols for use with the `JsonSchemaEncoder` and/or future encoder implementations.
+- `Json` and `Custom` have been added to supported formats in `SchemaFormat`.
+- `V2022_10` has been added to `ApiVersion` and set as the default API version.
```
Comment (kashifkhan) [inline sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md:13]: do you want to mention the behavior for unknown types here or in a sample?

File: sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md [hunk 2]
```diff
@@ -6,15 +6,16 @@ This version and all future versions will require Python 3.8+. Python 3.7 is no
...
 
 ### Bugs Fixed
 
 - Fixed a bug in sync/async `register_schema` and `get_schema_properties` that did not accept case insensitive strings as an argument to the `format` parameter.
+- Fixed a bug where unknown content type strings from the service raised a client error, rather than being returned as a string in the SchemaProperties `format` property.
```
Comment (kashifkhan) [inline sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md:18]: is there a sample that shows how to deal with this?

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_patch.py [hunk 3]
```diff
@@ -57,22 +58,24 @@ def _parse_schema_properties_dict(response_headers: Mapping[str, Union[str, int]
         "version": int(response_headers["Schema-Version"]),
     }
 
+def _normalize_content_type(content_type: str) -> str:
+    return content_type.replace(" ", "").lower()
```
Comment (kashifkhan) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_patch.py:62]: a content type can return as `a vro`

File: sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md (hunk 2 above)
Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md:18]: There isn't, but it feels like a niche scenario, so I didn't add it. Plus, I would hope to redirect people to the warning where we tell people to upgrade to the latest if they get back a string.

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_patch.py (hunk 3 above)
Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_patch.py:62]: i.e. These should be equivalent: application/json;serialization=Avro
vs `
application/json; serialization=Avro

File: sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md (hunk 2 above)
Comment (kashifkhan) [reply sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md:18]: ah fair enough :)

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**